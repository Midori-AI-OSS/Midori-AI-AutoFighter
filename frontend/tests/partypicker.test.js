import { describe, expect, test } from 'bun:test';
import { readFileSync } from 'fs';
import { join } from 'path';

describe('PartyPicker component', () => {
  test('uses dynamic assets and player fetch', () => {
    const content = readFileSync(join(import.meta.dir, '../src/lib/PartyPicker.svelte'), 'utf8');
    expect(content).toContain('import.meta.glob');
    expect(content).toContain('getPlayers');
    expect(content).toContain('mask-image');
  });
});
