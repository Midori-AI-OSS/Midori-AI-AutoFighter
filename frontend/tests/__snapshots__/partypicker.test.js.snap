// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`PartyPicker component roster layout snapshot 1`] = `
"<div class="roster-list">
      {#each roster as char}
        <button
          type="button"
          data-testid={\`choice-\${char.id}\`}
          class="char-row"
          class:selected={selected.includes(char.id)}
          on:click={() => select(char.id)}
          style={\`border-color: \${getElementColor(char.element)}\`}>
          <img src={char.img} alt={char.name} class="row-img" />
          <span class="row-name">{char.name}</span>
          <svelte:component
            this={getElementIcon(char.element)}
            class="row-type"
            style={\`color: \${getElementColor(char.element)}\`}
            aria-hidden="true" />
        </button>
      {/each}
    </div>"
`;
